plugins/machina-tools.js

```diff
@@ -0,0 +1,554 @@
+      machinaTools.displayLayer.addLayer(machinaTools.circleDisplayLayer);
+      $('.leaflet-control-layers-list span:contains("Machina Level Up Link Radius")').parent('label').removeClass('disabled').attr('title', '');
+    }
+  } else {
+    // Remove the circle layer from the display layer if necessary, and add the disabled mark.
+    if (machinaTools.displayLayer.hasLayer(machinaTools.circleDisplayLayer)) {
+      machinaTools.displayLayer.removeLayer(machinaTools.circleDisplayLayer);
+      $('.leaflet-control-layers-list span:contains("Machina Level Up Link Radius")')
+        .parent('label')
+        .addClass('disabled')
+        .attr('title', 'Zoom in to show those.');
+    }
+  }
+};
+
+machinaTools.showOrHideMachinaConflictArea = function () {
+  if (!machinaTools.conflictLayer.hasLayer(machinaTools.conflictAreaLayer)) {
+    machinaTools.conflictLayer.addLayer(machinaTools.conflictAreaLayer);
+    console.log('conflictAreaLayer activated');
+    $('.leaflet-control-layers-list span:contains("Machina Conflict Area)').parent('label').removeClass('disabled').attr('title', '');
+  } else {
+    machinaTools.conflictLayer.removeLayer(machinaTools.conflictAreaLayer);
+    console.log('conflictAreaLayer disabled');
+    $('.leaflet-control-layers-list span:contains("Machina Conflict Area")').parent('label').addClass('disabled').attr('title', 'Zoom in to show those.');
+  }
+};
+
+machinaTools.guessLevelByRange = function (linkLength) {
+  for (var level = 0; level <= 8; ++level) {
+    if (window.LINK_RANGE_MAC[level] >= linkLength) {
+      return level;
+    }
+  }
+  return 0;
+};
+
+machinaTools.linkAdded = function (data) {
+  data.link.on('add', function () {
+    // if (TEAM_NAMES[this.options.team] != undefined) {
+    if (window.TEAM_NAMES[this.options.team] === window.TEAM_NAME_MAC) {
+      var link = data.link.options.data;
+
+      // add destination portal - 1 level
+      machinaTools.drawExclusion(link.dGuid, 1, machinaTools.getDLatLng(link), true);
+
+      // add origin portal - level based on link length
+      var linkLength = machinaTools.getLinkLength(link);
+      var level = machinaTools.guessLevelByRange(linkLength);
+      machinaTools.drawExclusion(link.oGuid, level, machinaTools.getOLatLng(link), true);
+    }
+    // }
+  });
+};
+
+function humanFileSize(size) {
+  var i = size === 0 ? 0 : Math.floor(Math.log(size) / Math.log(1024));
+  return (size / Math.pow(1024, i)).toFixed(2) * 1 + ' ' + ['B', 'kB', 'MB', 'GB', 'TB'][i];
+}
+
+machinaTools.showConflictAreaInfoDialog = function () {
+  var html = $('<div>');
+  if (machinaTools.conflictAreaLast && machinaTools.conflictAreaLast.getLayers().length > 0) {
+    var ul = $('<ul>');
+    ul.appendTo(html);
+
+    var conflictAreaLayers = machinaTools.conflictAreaLast.getLayers();
+    var conflictLayerGeometry = conflictAreaLayers[0].feature.geometry;
+    ul.append($('<li>', { html: `Cluster area size: ${humanFileSize(sizeof(conflictLayerGeometry))}` }));
```

Comment:
about 16 x totalPoints (+ 60 for the keys/strings), sizeof doesn't count JS structures (array/object)

Reply:
replaced with implementation based on https://stackoverflow.com/questions/1248302/how-to-get-the-size-of-a-javascript-object#29387295